From ccfbdb515c4ee3e027f7452da14bff7d372dafd3 Mon Sep 17 00:00:00 2001
From: sbruno <sbruno@FreeBSD.org>
Date: Sun, 23 Sep 2012 20:24:17 +0000
Subject: [PATCH 7/9] MFC r238371

When an MFI command fails, the driver needs to set bio->bio_resid so that
the upper levels notice.  Otherwise we see commands silently failing leading
to data corruption.  This mirrors dadone()

Submitted by: Andrew Boyer aboyer@averesystems.com
---
 sys/dev/mfi/mfi_disk.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sys/dev/mfi/mfi_disk.c b/sys/dev/mfi/mfi_disk.c
index fc43e67..18ceb99 100644
--- a/sys/dev/mfi/mfi_disk.c
+++ b/sys/dev/mfi/mfi_disk.c
@@ -298,6 +298,7 @@ mfi_disk_complete(struct bio *bio)
 	hdr = bio->bio_driver1;
 
 	if (bio->bio_flags & BIO_ERROR) {
+		bio->bio_resid = bio->bio_bcount;
 		if (bio->bio_error == 0)
 			bio->bio_error = EIO;
 		disk_err(bio, "hard error", -1, 1);
-- 
1.7.10.4

